### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by jtwhite79**
pulled up updated message passing for yamr from Aymans fork, update pestpp-opt to handle (slightly) out of bounds optimal dec vars, added some checking for base_jco compatibility in pestpp-glm

### CODE REVIEW

File: src/libs/run_managers/abstract_base/model_interface.cpp [hunk 1]
```diff
@@ -990,7 +991,6 @@ string InstructionFile::parse_obs_name_from_token(const string& token)
 		}
 	}
 	throw_ins_error("instruction type not recognized for observation instruction '" + token + "'");
-	return "";
```
Comment (mwtoews) [inline src/libs/run_managers/abstract_base/model_interface.cpp:993]: undo these; see f690ffbb63cff2dce5c4a77af0609dd1c21aa5f0

File: src/libs/run_managers/abstract_base/abstract_base.vcxproj [hunk 2]
```diff
@@ -138,7 +138,7 @@
     <ClCompile Include="model_interface.cpp" />
     <ClCompile Include="RunManagerAbstract.cpp" />
     <ClCompile Include="RunStorage.cpp" />
-    <ClCompile Include="Serialization.cpp" />
+    <ClCompile Include="Serializeation.cpp" />
```
Comment (mwtoews) [inline src/libs/run_managers/abstract_base/abstract_base.vcxproj:141]: also Serializeation -> Serialization

File: src/libs/common/network_package.cpp [hunk 3]
```diff
@@ -22,7 +22,7 @@ bool NetPackage::allowable_ascii_char(int8_t value)
 	// This is for security resasons.  Only \0
 	//  and printable characters are allowed
 	bool ret_val = false;
-	if (value == 0 || (value >= 32 && value <= 126))
+	if (value == 0 || (value >= 32 && value <= 176))
```
Comment (mwtoews) [inline src/libs/common/network_package.cpp:25]: undo this edit, see a8f3e0f2a80a9d3f49e7571d6ecdd95d029d1e1a

File: src/libs/run_managers/abstract_base/RunStorage.cpp [hunk 4]
```diff
@@ -515,10 +515,13 @@ int RunStorage::get_run(int run_id, double *pars, size_t npars, double *obs, siz
...
 
-	if (npars != p_size) {
+	assert(npars == p_size);
+	assert(nobs == o_size);
+
+	if (npars != npars) {
 		throw(PestIndexError("RunStorage::get_run: parameter dimension in incorrect"));
 	}
-	if (nobs != o_size) {
+	if (nobs != nobs) {
```
Comment (mwtoews) [inline src/libs/run_managers/abstract_base/RunStorage.cpp:524]: undo a few lines here; see 73b1600bdff27c5aa9efeef02b50970be519c550

File: src/libs/run_managers/abstract_base/Makefile [hunk 5]
```diff
@@ -8,7 +8,7 @@ OBJECTS := \
     model_interface \
     RunManagerAbstract \
     RunStorage \
-    Serialization
+    Serializeation
```
Comment (mwtoews) [inline src/libs/run_managers/abstract_base/Makefile:11]: also Serializeation -> Serialization

File: src/libs/run_managers/abstract_base/CMakeLists.txt [hunk 6]
```diff
@@ -7,7 +7,7 @@ add_library(rm_abstract STATIC
   model_interface.cpp
   RunManagerAbstract.cpp
   RunStorage.cpp
-  Serialization.cpp
+  Serializeation.cpp
```
Comment (mwtoews) [inline src/libs/run_managers/abstract_base/CMakeLists.txt:10]: undo these changes (a few more further down too); see 909edb3124a6cc46ae496ab35fd4301129d26977

File: src/libs/run_managers/abstract_base/abstract_base_intel.vcxproj [hunk 7]
```diff
@@ -149,7 +149,7 @@
     <ClCompile Include="model_interface.cpp" />
     <ClCompile Include="RunManagerAbstract.cpp" />
     <ClCompile Include="RunStorage.cpp" />
-    <ClCompile Include="Serialization.cpp" />
+    <ClCompile Include="Serializeation.cpp" />
```
Comment (mwtoews) [inline src/libs/run_managers/abstract_base/abstract_base_intel.vcxproj:152]: also Serializeation -> Serialization

File: src/libs/run_managers/yamr/PantherAgent.cpp [hunk 8]
```diff
@@ -330,18 +336,20 @@ NetPackage::PackType PANTHERAgent::run_model(Parameters &pars, Observations &obs
...
+		smessage << ex.what();
+		NetPackage::PackType::RUN_FAILED;
+		
 	}
 	catch(...)
 	{
  		cerr << "   Error running model" << endl;
 		cerr << "   Aborting model run" << endl;
-		final_run_status = NetPackage::PackType::RUN_FAILED;
+		NetPackage::PackType::RUN_FAILED;
```
Comment (mwtoews) [inline src/libs/run_managers/yamr/PantherAgent.cpp:347]: probably better to assign `NetPackage::PackType::RUN_FAILED` somewhere (as it was with 1913e4986be18df7564fb1f3fe0acd851f98a8f0), otherwise this line does nothing.

File: src/libs/run_managers/abstract_base/Serializeation.cpp [hunk 9]
```diff
@@ -0,0 +1,280 @@
+/*
```
Comment (mwtoews) [inline src/libs/run_managers/abstract_base/Serializeation.cpp:1]: this file slipped in when renaming Serializeation.cpp -> Serialization.cpp; see 909edb3124a6cc46ae496ab35fd4301129d26977 ; were there any changes?

**Comment by jtwhite79** [comment]
Thanks @mwtoews for catching this. Im gonna leave the misspelled serialization cpp for now since there are a few forks under active dev and I dont wanna goof things up for them, esp since at least one of them is making mods to the run mgr process
